### PR TITLE
Create CLI reference with redirect, update TOCs

### PR DESCRIPTION
**Pull request by lorikrell**
## Purpose of this PR

Add MSI Inventory Management CLI Reference for MSI 1.1.2.

Fixes MSI Issue: https://github.com/magento-engcom/msi/issues/2143

## Affected URLs

Replacing current CLI page with a new reference, added redirect, updated TOCs.

- redirected/removed page: https://devdocs.magento.com/guides/v2.3/config-guide/cli/config-cli-inventory.html
- new: https://devdocs.magento.com/guides/v2.3/inventory/inventory-cli-reference.html

## Links to source code

- https://github.com/magento-engcom/msi/tree/2.3-develop/app/code/Magento/InventoryReservationCli/Command

whatsnew
Added a new [Inventory CLI reference](https://devdocs.magento.com/guides/v2.3/inventory/inventory-cli-reference.html) for Inventory Management including new commands for reservation updates.

### CODE REVIEW

**Comment by lorikrell** [comment]
Feedback from Igor:

```
Inconsistencies in reservations may occur in the following situations:

- Mass transfering inventory
- Removing sources from stocks
- Reassigning websites to different stocks
```

I would better to describe these scenarios in more details. Currently, from what I've seen on stackexchange and those guys who approached me usually face next issues:
https://magento.stackexchange.com/questions/268853/salable-quantity-is-greater-than-quantity/269711#269711

- Upgrade Magento to 2.3.* with orders not in Finite State (Complete, Canceled, Closed). Because for these orders Magento with MSI will create a compensational reservation. So that it means that there would not be initial reservation (deducting one), but we will have a compensational one.
- If merchant used "Manage Stock = false" configuration, Magento does not place reservations at both order placement and shipment events. Thus, if merchant changed the Manage Stock configuration and some orders were created at that time, it means that after orders would be handled the Salable Qty would be corrupted with compensation reservation.
- If Merchant will re-assign the Stock for Website and there are created orders placed on that website, the compensational reservation would be created for the new stock, while the initial reservation created for previously assigned Stock.
- If for some reasons we have SUM among all the reservation quantities created in the scope of an Order ( which is currently in finite state (Complete, Canceled, Closed)) different from `0`

File: guides/v2.3/inventory/inventory-cli-reference.md [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+- Product bike-123 should be compensated by +2.000000 for stock 1
+```
+
+```text
+172:bike-123:+2.000000:1
+```
+
+### Create compensations command
+
+After running `inventory:reservation:list-inconsistencies`, the `create-compensations` command creates compensation reservations using the returned list of inconsistencies. Depending on the issue, new reservations are created to either place or release a hold on salable quantity.
```
Comment (keharper) [inline guides/v2.3/inventory/inventory-cli-reference.md]:
```suggestion
The `create-compensations` command creates compensation reservations for the issues detected by the  `list-inconsistencies` command. Depending on the issue, new reservations are created to either place or release a hold on salable quantity.
```

File: guides/v2.3/inventory/inventory-cli-reference.md [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+
+- `-c`, `--complete-orders` - Creates reservations for only completed order inconsistencies.
+- `-i`, `--incomplete-orders` - Creates reservations for only incomplete order inconsistencies.
+- `-r`, `raw` - Raw output.
+
+Requested compendations must be provided using this format:  `<ORDER_INCREMENT_ID>:<SKU>:<QUANTITY>:<STOCK-ID>`.
+
+If the format of the request is incorrect, the following message displays: A list of compensations needs to be defined as argument or STDIN.
+
+As reservations are entered, messages display indicating the updates by SKU, order, and stock.
```
Comment (keharper) [inline guides/v2.3/inventory/inventory-cli-reference.md]:
```suggestion
As the command creates reservations, it displays messages indicating the updates by SKU, order, and stock.
```

File: _data/toc/inventory.yml [hunk 3]
```diff
@@ -14,12 +14,46 @@ pages:
...
     - label: Deprecated CatalogInventory entities
       url: /inventory/catalog-inventory-replacements.html
 
     - label: Deprecated ScalableInventory entities
       url: /inventory/scalable-inventory-replacements.html
 
+    - label: Order processing with Inventory Management
+      class: tutorial
+      url: /rest/tutorials/inventory/index.html
+
```
Comment (keharper) [inline _data/toc/inventory.yml]: Delete this empty line

File: guides/v2.3/inventory/inventory-cli-reference.md [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+Inventory Management provides CLI commands to manage inventory data and configuration settings.
+
+These commands include:
+
+- Checking and resolving reservation inconsistencies affecting salable quantity
+- Adding geocodes for the Distance Priority algorithm
+
+## Resolve reservations inconsistencies
+
+[Reservations]({{ page.baseurl }}/rest/tutorials/inventory/reservations.html) place a salable quantity hold for product SKUs per stock. When you ship, add products, cancel, or refund an order, compensation reservations enter to place or clear these holds.
```
[the remark this answers is not in the document]
Comment (lorikrell) [reply guides/v2.3/inventory/inventory-cli-reference.md]: Weird copy paste, thanks!

File: guides/v2.3/inventory/inventory-cli-reference.md [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+
+### Create compensations command
+
+After running `inventory:reservation:list-inconsistencies`, the `create-compensations` command creates compensation reservations using the returned list of inconsistencies. Depending on the issue, new reservations are created to either place or release a hold on salable quantity.
+
+```bash
+bin/magento inventory:reservation:create-compensations
+```
+
+Command options:
```
[the remark this answers is not in the document]
Comment (lorikrell) [reply guides/v2.3/inventory/inventory-cli-reference.md]: added!

File: guides/v2.3/inventory/inventory-cli-reference.md [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+- Product bike-123 was compensated by +2.000000 for stock 1
+- Product bikehat-456 was compensated by +1.000000 for stock 1
+```
+
+After updates complete, run the list command to verify:
+
+```bash
+bin/magento inventory:reservation:list-inconsistencies -r
+```
+
```
[the remark this answers is not in the document]
Comment (lorikrell) [reply guides/v2.3/inventory/inventory-cli-reference.md]: got it!

File: guides/v2.3/inventory/inventory-cli-reference.md [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+```bash
+bin/magento inventory:reservation:list-inconsistencies
+```
+
+Command options:
+
+- `-c`, `--complete-orders` - Returns inconsistencies for only completed orders. Incorrect reservations may still be on hold for completed orders.
+- `-i`, `--incomplete-orders` - Returns inconsistencies for only incomplete orders (partially shipped, unshipped). Incorrect reservations may hold too much or not enough salable quantity for the orders.
+- `-r`, `--raw` - Returns inconsistences for all orders in raw output.
```
[the remark this answers is not in the document]
Comment (lorikrell) [reply guides/v2.3/inventory/inventory-cli-reference.md]: Yes, fixing content. -r is for the raw output instead of prettified.

**Comment by lorikrell** [comment]
All comments updated, commands verified, and added content per discussion with dev.

**Pull request merged**